packages/medusa/src/interfaces/payment-processor.ts

```diff
@@ -0,0 +1,220 @@
+import {
+  Address,
+  Customer,
+  PaymentSessionStatus,
+  ShippingMethod,
+} from "../models"
+import { MedusaContainer } from "../types/global"
+
+export type PaymentProcessorContext = {
+  shipping_address?: Address | null
+  shipping_methods?: ShippingMethod[]
+  billing_address?: Address | null
+  email: string
+  currency_code: string
+  amount: number
+  resource_id?: string
+  customer?: Customer
+}
+
+export type PaymentProcessorSessionResponse = {
+  update_requests: { customer_metadata: Record<string, unknown> }
+  session_data: Record<string, unknown>
+}
+
+export interface PaymentProcessorError {
+  error: string
+  code: number
+  details: any
+}
+
+/**
+ * The new payment service plugin interface
+ */
+export interface PaymentProcessor {
+  /**
+   * Return a unique identifier to retrieve the payment plugin provider
+   */
+  getIdentifier(): string
+
+  /**
+   * Used to initialise anything like an SDK or similar
+   */
+  init(): Promise<void>
```

Comment:
question: at what point in the life cycle does this get called?

Reply:
I thought calling it as part of either the loader or the install provider. wdyt?

Reply:
As this would also need to be added to the payment plugin registration, I would keep it for a separate PR. Perhaps the one where we address the other topic.

So should we leave it out for now?

Reply:
Here the purpose was to describe the API. everything related to the loader etc will be made in the next pr @olivermrbl